packages/@aws-cdk/aws-lambda-golang/README.md

```diff
@@ -0,0 +1,249 @@
+The [aws-lambda-go](https://github.com/aws/aws-lambda-go) library has built in support for the provided runtime as long as
+you name the handler `bootstrap` (which we do by default).
+
+## Dependencies
+
+The construct will attempt to figure out how to handle the dependencies for your function. It will
+do this by determining whether or not you are vendoring your dependencies. It makes this determination
+by looking to see if there is a `vendor` folder at the [project root](#project-root).
+
+With this information the construct can determine what commands to run. You will
+generally fall into two scenarios:
+
+1. You are using vendoring (indicated by the presence of a `vendor` folder)
+   In this case `go build` will be run with `-mod=vendor` set
+2. You are not using vendoring (indicated by the absence of a `vendor` folder)
+   If you are not vendoring then `go build` will be run without `-mod=vendor`
+    since the default behavior is to download dependencies
+
+All other properties of `lambda.Function` are supported, see also the [AWS Lambda construct library](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-lambda).
+
+## Environment
+
+By default the following environment variables are set for you:
+
+* `GOOS=linux`
+* `GOARCH=amd64`
+* `GO111MODULE=on`
+
+Use the `environment` prop to define additional environment variables when go runs:
+
+```ts
+new lambda.GolangFunction(this, 'handler', {
+  entry: 'app/cmd/api',
+  bundling: {
+    environment: {
+      HELLO: 'WORLD',
+    },
+  },
+});
+```
+
+## Docker
+
+To force bundling in a docker container, set the `forceDockerBundling` prop to `true`. This
+
+Use the `buildArgs` prop to pass build arguments when building the bundling image:
+
+```ts
+new lambda.GolangFunction(this, 'handler', {
+  entry: 'app/cmd/api',
+  bundling: {
+    buildArgs: {
+      HTTPS_PROXY: 'https://127.0.0.1:3001',
+    },
+  },
+});
+```
+
+Use the `bundlingDockerImage` prop to use a custom bundling image:
+
+```ts
+new lambda.GolangFunction(this, 'handler', {
+  entry: 'app/cmd/api',
+  bundling: {
+    dockerImage: dk.BundlingDockerImage.fromAsset('/path/to/Dockerfile'),
+  },
+});
+```
+
+Use the `goBuildFlags` prop to pass additional build flags to `go build`:
+
+```ts
+new lambda.GolangFunction(this, 'handler', {
+  entry: 'app/cmd/api',
+  bundling: {
+    goBuildFlags: ['-ldflags "-s -w"'],
+  },
+});
+```
+
+## Command hooks
+
+It is  possible to run additional commands by specifying the `commandHooks` prop:
+
+```ts
+new lambda.GolangFunction(this, 'handler', {
+  bundling: {
+    commandHooks: {
+      // run tests
+      beforeBundling(inputDir: string): string[] {
+        return ['go test ./cmd/api -v'];
+      },
+      // ...
+    },
+  },
+});
+```
+
+The following hooks are available:
+
+* `beforeBundling`: runs before all bundling commands
+* `beforeInstall`: does not run
```

Comment:
is not implemented

```suggestion
```

Reply:
@jogold I removed this, but I had originally left it there to remind myself to ask you about this. I remember a comment on the PR where you added commandHooks that mentioned making these mandatory so that all languages would have a similar experience. https://github.com/aws/aws-cdk/pull/11583#discussion_r530391166

Reply:
You can safely remove it. The discussion was about making the methods in `ICommandHooks` mandatory or not so that the experience in all the _JSII translated languages_ would be the same.

Reply:
Ok cool.